File: src/domain/diary/diary-entry.ts
import type { Nutrition } from '@/domain/nutrition/nutrition'

export type MealType = 'breakfast' | 'lunch' | 'dinner' | 'snack'
export type DiarySourceType = 'product' | 'recipe'

/**
 * Nutrition is a snapshot captured when the entry is added and must never be
 * recalculated automatically after a product or recipe version changes.
 */
export interface DiaryEntry extends Nutrition {
  id: string
  date: string
  mealType: MealType
  /** Persistent position within one date and meal section. */
  sortOrder: number
  sourceType: DiarySourceType
  sourceId: string
  sourceVersionId: string
  sourceName: string
  amount: number
  unit: string
  createdAt: string
  updatedAt: string
  deletedAt?: string
}

File: ios/Domain/Diary/DiaryEntry.swift
import Foundation

struct DiaryContext: Hashable, Codable, Sendable {
    let day: LocalDay
    let meal: MealType
}

struct FoodSourceReference: Hashable, Codable, Sendable {
    let sourceType: SourceType
    let sourceID: UUID
}

/// Nutrition and source fields are snapshots. They must not be automatically recalculated.
struct DiaryEntry: Identifiable, Hashable, Codable, Sendable {
    let id: UUID
    let day: LocalDay
    let mealType: MealType
    let sortOrder: Int
    let sourceType: SourceType
    let sourceID: UUID
    let sourceVersionID: UUID
    let sourceName: String
    let amount: Double
    let unitToken: String
    let nutrition: Nutrition
    let createdAt: Date
    let updatedAt: Date
    let deletedAt: Date?
}
